code/game/objects/items/tcg/tcg.dm

```diff
@@ -71,6 +71,121 @@ GLOBAL_LIST_EMPTY(cached_cards)
+	if(istype(I, /obj/item/tcgcard))
+		var/obj/item/tcgcard/second_card = I
+		var/obj/item/tcgcard_deck/new_deck = new /obj/item/tcgcard_deck(drop_location())
+		new_deck.flipped = flipped
+		user.transferItemToLoc(second_card, new_deck)//Start a new pile with both cards, in the order of card placement.
+		user.transferItemToLoc(src, new_deck)
+		new_deck.update_icon_state()
+	return ..()
+
+/**
+  * A stack item that's not actually a stack because ORDER MATTERS with a deck of cards!
+  * The "top" card of the deck will always be the bottom card in the stack for our purposes.
+  */
+/obj/item/tcgcard_deck
+	name = "Trading Card Pile"
+	desc = "A stack of TCG cards."
+	icon = DEFAULT_TCG_DMI_ICON
+	icon_state = "deck_up"
+	obj_flags = UNIQUE_RENAME
+	var/flipped = FALSE
+	var/static/radial_draw = image(icon = 'icons/mob/radial.dmi', icon_state = "radial_draw")
+	var/static/radial_shuffle = image(icon = 'icons/mob/radial.dmi', icon_state = "radial_shuffle")
+	var/static/radial_pickup = image(icon = 'icons/mob/radial.dmi', icon_state = "radial_pickup")
+
+/obj/item/tcgcard_deck/Initialize()
+	. = ..()
+	LoadComponent(/datum/component/storage/concrete/tcg)
+
+/obj/item/tcgcard_deck/update_icon_state()
+	. = ..()
+	if(flipped)
+		icon_state = "deck_down"
+	else
+		icon_state = "deck_up"
+
+/obj/item/tcgcard_deck/examine(mob/user)
+	. = ..()
+	. += "<span class='notice'>\The [src] has [contents.len] cards inside.</span>"
+
+/obj/item/tcgcard_deck/attack_hand(mob/user)
+	var/list/choices = list(
+		"Draw" = image(icon = 'icons/mob/radial.dmi', icon_state = "radial_draw"),
+		"Shuffle" = image(icon = 'icons/mob/radial.dmi', icon_state = "radial_shuffle"),
+		"Pickup" = image(icon = 'icons/mob/radial.dmi', icon_state = "radial_pickup"),
+		"Flip" = image(icon = 'icons/mob/radial.dmi', icon_state = "radial_flip"),
+		)
+	var/choice = show_radial_menu(user, src, choices, custom_check = CALLBACK(src, .proc/check_menu, user), require_near = TRUE, tooltips = TRUE)
+	if(!check_menu(user))
+		return
+	switch(choice)
+		if("Draw")
+			draw_card(user)
+		if("Shuffle")
+			shuffle_deck(user)
+		if("Pickup")
+			user.put_in_hands(src)
+		if("Flip")
+			flipped = !flipped
+			contents = reverseRange(contents)
+			update_icon_state()
+		if(null)
+			return
+
+/obj/item/tcgcard_deck/proc/check_menu(mob/living/user)
+	if(!istype(user))
+		return FALSE
+	if(user.incapacitated() || !user.Adjacent(src))
+		return FALSE
+	return TRUE
+
+/obj/item/tcgcard_deck/attackby(obj/item/I, mob/living/user, params)
+	. = ..()
+	if(istype(I, /obj/item/tcgcard))
+		if(contents.len >= 30)
+			to_chat(user, "<span class='notice'>This pile has too many cards for a regular deck!</span>")
+			return FALSE
+		var/obj/item/tcgcard/new_card = I
+		new_card.flipped = flipped
+		user.transferItemToLoc(new_card, src)
+
+/obj/item/tcgcard_deck/proc/draw_card(mob/user)
+	if(!contents.len)
+		CRASH("A TCG deck was created with no cards inside of it.")
```

Comment:
CRASH kills the current proc, think of it like a runtime. If you want to continue execution after it runs, use stack_trace()